webapp/admin/maintenance.py

```
@@ -626,7 +630,90 @@ def optimize_database():
+            db.session.add(log_entry)
+            db.session.commit()
+
+            logger.warning("Environment configuration updated. Restart required for changes to take effect.")
+
+            return jsonify({
+                "message": "Configuration updated successfully. Restart the application for changes to take effect.",
+                "backup_path": str(backup_path) if backup_path else None,
+                "restart_required": True,
+            })
+
+        except Exception as exc:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)